### DISCUSSION TITLE
Welcome to Clp Discussions!

### DISCUSSION THREAD

**Discussion by tkralphs** [coin-or/Clp#168] [General]
## 👋 Welcome!
  As you may have heard, Github has been developing a new feature called Discussions, which is now being rolled 
  out in beta to all projects. We envision Discussions as a way of moving the kinds of conversations that currently 
  take place on the old Mailman list serve directly to where development is taking place! To take this thing for a test 
  drive, please post here when you would otherwise post to the list serve. This is a place to connect with other members 
  of our community on topics not suited for opening an Issue. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, feel free to comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ralacroix** [comment]
I would like to see a way to stop CLP after a given amount of time (clock time or CPU time) has elapsed, and return with the best solution so far and a status indicating that CLP ran out of time. We use Clp in a production environment, with a periodic calibration run that does thousands of similar but independent LPs, usually scheduled for a weekend and taking all available cores (~ 100). Most of the LPs solve in under 2 minutes but several take over half an hour of computation to return a solution. Today I killed one LP that was still running after using over 10 hours of CPU time without returning from the call to CLP Simplex solver. Solving this particular LP with the Barrier method returned a solution in under 5 minutes. Currently the only way I see of implementing this time limit is by imposing an external CPU time limit on the process performing the computation and then detecting that the operating system killed the process when it has used up its time quota. This is not an ideal (portable) solution. Portability is highly desired, since this mission-critical application is over 30 years old and has survived porting across 4 different CPU architectures (currently running on x86-64-bit Linux). It started using CLP as the solver around 8 years ago.
